src/Mono.Android/Java.Interop/TypeManager.cs

```diff
@@ -253,14 +253,29 @@ static Type monovm_typemap_java_to_managed (string java_type_name)
 
 		internal static Type? GetJavaToManagedType (string class_name)
 		{
-			Type? type = JNIEnvInit.RuntimeType switch {
+			lock (TypeManagerMapDictionaries.AccessLock) {
+				return GetJavaToManagedTypeCore (class_name);
+			}
+		}
+
+		private static Type? GetJavaToManagedTypeCore (string class_name)
+		{
+			Type? type = null;
+
+			if (TypeManagerMapDictionaries.JniToManaged.TryGetValue (class_name, out type)) {
```

Comment:
`type` can be declared inline, removing the need to initialize its value above.

```suggestion
			if (TypeManagerMapDictionaries.JniToManaged.TryGetValue (class_name, out Type? type)) {
```